biorxiv/article_distances/04_biorxiv_article_distances_abstract_only.py

```diff
@@ -0,0 +1,239 @@
+                "pmcid": (
+                    PMC_off_published.query(f"journal=='{row['journal']}'")
+                    .reset_index()
+                    .document.values[0]
+                ),
+                "journal": row["journal"],
+                "distance": off_article_dist[idx, journal_mapper[row["journal"]]],
+            }
+        )
+
+final_df = articles_distance_df.assign(label="pre_vs_published").append(
+    pd.DataFrame.from_records(data).assign(label="pre_vs_random")
+)
+final_df.head()
+
+final_df = biorxiv_journal_df[["document", "preprint_doi"]].merge(final_df)
+final_df.to_csv(
+    "output/annotated_links/article_distances_abstract_only.tsv", sep="\t", index=False
+)
+final_df.head()
+
+# # Distribution plot
+
+g = (
+    p9.ggplot(
+        final_df.replace(
+            {
+                "pre_vs_published": "preprint-published",
+                "pre_vs_random": "preprint-random",
+            }
+        )
+    )
+    + p9.aes(x="label", y="distance")
+    + p9.geom_violin(fill="#a6cee3")
+    + p9.labs(x="Document Pair Groups", y="Euclidean Distance")
+    + p9.theme_seaborn(context="paper", style="ticks", font="Arial", font_scale=1.35)
+)
+g.save("output/figures/biorxiv_article_distance_abstract_only.svg", dpi=250)
+g.save("output/figures/biorxiv_article_distance_abstract_only.png", dpi=250)
+print(g)
+
+# # Plot Abstract Only vs Full Text Only
+
+abstract_only = final_df
+full_text = pd.read_csv("output/annotated_links/article_distances.tsv", sep="\t")
+
+plot_df = (
+    full_text.query("label=='pre_vs_published'")
+    .rename(index=str, columns={"distance": "full_text_distance"})[
+        ["document", "full_text_distance"]
+    ]
+    .merge(
+        abstract_only.query("label=='pre_vs_published'").rename(
+            index=str, columns={"distance": "abstract_only_distance"}
+        )[["document", "abstract_only_distance"]],
+        on="document",
+    )
+    .assign(
+        abstract_only_distance_log10=lambda x: -np.log10(x.abstract_only_distance),
+        full_text_distance_log10=lambda x: -np.log10(x.full_text_distance),
+    )
+)
+plot_df.head()
+
+g = (
+    p9.ggplot(plot_df)
+    + p9.aes(x="full_text_distance_log10", y="abstract_only_distance_log10")
+    + p9.geom_point(fill="#a6cee3")
+    + p9.xlim([-1, 6])
+    + p9.labs(x="Full Text Distance (-log 10)", y="Abstract Only Distance (-log 10)")
+    + p9.theme_seaborn(context="paper", style="ticks", font="Arial", font_scale=1.35)
+)
+g.save("output/figures/biorxiv_full_text_v_abstract_only.svg", dpi=250)
+g.save("output/figures/biorxiv_full_text_v_abstract_only.png", dpi=250)
+print(g)
+
+plot_df.sort_values("abstract_only_distance_log10", ascending=False).head(10)
+
+# The negative values for abstract only consist of high confidence scores for a true match. The table for the first few entries have an abstract only distance of practically zero. This means the log version will be highly negative.
```

Comment:
Do you have an intution for what the cluster at the bottom is? Are those just articles whose abstract changed completely when they were published?

Reply:
This description is wrong and will be changed. Turns out I took the negative log of the distances so what I wrote is misleading. 

Each point within this plot are preprints matched with their corresponding published article. The intuition here is to see if abstract distances correlate with full text distances. If it does, then that would mean abstracts alone can be used to match documents. The blob below is majority of the articles while the outliers above are papers whose abstracts didn't change very much or at all. (will manually confirm) Long story short: I'll be updating this figure and description so everything makes more sense.